pyrallis/utils.py

```diff
@@ -572,6 +574,60 @@ def deflatten(d: Dict[str, Any], sep: str = '.'):
     return deflat_d
 
 
+def parse_string(s):
```

Comment:
Also, add type hints to the 3 new functions

Reply:
Moving to a dedicated file sounds like a good idea.
A dedicated parser for each new format doesn't seem to be worth the trouble here, but there might be a better solution that still follows the OCP. Will give it some thought 🧠

Reply:
@idow09 would you say that a dispatch-based mechanism would be better here?
Something like
```python
@configdispatch
def parse_string(s):
    raise RuntimeError(f'Unhandled config type')

@parse_string.register(ConfigType.YAML)
def parse_string_yaml(s):
    import yaml
    return yaml.safe_load(s)


@parse_string.register(ConfigType.JSON)
def parse_string_json(s):
    import json
    try:
        return json.loads(s)
    except json.decoder.JSONDecodeError:
        return s  # No parsing to be done, simply return value
```

IMO it would make sense if we had many supported configs, but right now I'm not sure it's better than the explicit function
```python
def parse_string(s):
    if Options.get_config_type() is ConfigType.YAML:
        import yaml
        return yaml.safe_load(s)
    elif Options.get_config_type() is ConfigType.JSON:
        import json
        try:
            return json.loads(s)
        except json.decoder.JSONDecodeError:
            return s # No parsing to be done, simply return value
    else:
        raise RuntimeError('Unhandled config type')
```

Reply:
Simplified by using Parser classes that are matched to the enum directly
```python
class ConfigType(ParserEnum):
    YAML = YAMLParser
    JSON = JSONParser
    TOML = TOMLParser
```